### PR TITLE
Task 8

### PR DESCRIPTION
**Pull request by SamoylenkVi**

### CODE REVIEW

File: project/src/components/AddFilmFavoriteButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { selectFavoriteFilms } from '../../store/favorite-films/selectors';
+
+type AddFilmFavoriteButtonProps = {
+  id: string;
+}
+
+export const AddFilmFavoriteButton = ({id}:AddFilmFavoriteButtonProps) => {
+  const dispatch = useAppDispatch();
+
+  const selectIsFilmFavorite = createSelectFilmByFilmId(id);
```
Comment (Pavel-Liteiniy) [inline project/src/components/AddFilmFavoriteButton/index.tsx]:
```suggestion
  const selectFilmByFilmId = createSelectFilmByFilmId(id);
```

Comment (Pavel-Liteiniy) [reply project/src/components/AddFilmFavoriteButton/index.tsx]: Or 
```suggestion
  const selectIsFilmFavorite = createSelectIsFilmFavoriteByFilmId(id);
```

File: project/src/components/Player/index.tsx [hunk 2]
```diff
@@ -1,34 +1,102 @@
...
+      video.addEventListener('timeupdate', updateFilmTime);
+      video.addEventListener('durationchange', updateDuration);
+    }
+
+    return () => {
+      video?.removeEventListener('timeupdate', updateFilmTime);
+      video?.removeEventListener('durationchange', updateDuration);
+    };
+
+  }, [videoRef]);
```
Comment (Pavel-Liteiniy) [inline project/src/components/Player/index.tsx]:
```suggestion
  }, []);
```

https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

File: project/src/index.tsx [hunk 3]
```diff
@@ -6,10 +6,11 @@ import { ToastContainer } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 import { store } from './store/store';
 import { MainPage } from './pages/MainPage';
+import { MyListPage } from './pages/MyListPage';
 import { NotFoundPage } from './pages/NotFoundPage';
 import { FilmDetailsPage } from './pages/FilmDetailsPage';
 import { AddReviewPage } from './pages/AddReviewPage';
-import { Player } from './components/Player';
+// import { Player } from './components/Player';
```
Comment (Pavel-Liteiniy) [inline project/src/index.tsx]: Commented code

File: project/src/index.tsx [hunk 4]
```diff
@@ -52,13 +57,13 @@ const router = createBrowserRouter([
...
         </PrivateRouter>,
       },
-      {
-        path: APP_ROUTE.Player,
-        element:
-        <PrivateRouter>
-          <Player/>
-        </PrivateRouter>,
-      },
+      // {
```
Comment (Pavel-Liteiniy) [inline project/src/index.tsx]: Commented code

File: project/src/pages/FilmDetailsPage/index.tsx [hunk 5]
```diff
@@ -37,6 +40,10 @@ export const FilmDetailsPage = () => {
     return null;
   }
 
+  if (!id) {
```
Comment (Pavel-Liteiniy) [inline project/src/pages/FilmDetailsPage/index.tsx]: There is a chance that `id` is not defined?

**Review by Pavel-Liteiniy — CHANGES_REQUESTED** [review]

File: project/src/components/AddFilmFavoriteButton/index.tsx (hunk 1 above)
Comment (SamoylenkVi) [reply project/src/components/AddFilmFavoriteButton/index.tsx]: fixed

File: project/src/components/Player/index.tsx (hunk 2 above)
Comment (SamoylenkVi) [reply project/src/components/Player/index.tsx]: fixed

File: project/src/index.tsx (hunk 3 above)
Comment (SamoylenkVi) [reply project/src/index.tsx]: fixed

File: project/src/index.tsx (hunk 4 above)
Comment (SamoylenkVi) [reply project/src/index.tsx]: fixed

File: project/src/pages/FilmDetailsPage/index.tsx (hunk 5 above)
Comment (SamoylenkVi) [reply project/src/pages/FilmDetailsPage/index.tsx]: fixed

File: project/src/pages/FilmDetailsPage/index.tsx [hunk 6]
```diff
@@ -75,10 +71,12 @@ export const FilmDetailsPage = () => {
               </p>
 
               <div className="film-card__buttons">
-                <PlayFilmButton id={Number(id)}/>
-                <AddFilmFavoriteButton id={id}/>
                 {(id !== undefined) && (
```
Comment (Pavel-Liteiniy) [inline project/src/pages/FilmDetailsPage/index.tsx]: Let's remove this `(id !== undefined) &&`

And use this instead `const { id } = useParams() as { id: string };`

**Review by Pavel-Liteiniy — APPROVED** [review]

**Comment by Pavel-Liteiniy** [comment]
Approve. Just fix this one https://github.com/SamoylenkVi/wtw_cinema/pull/6#discussion_r1301055883

**Pull request merged**